src/main/java/org/commonjava/indy/service/archive/jaxrs/ArchiveManageResources.java

```diff
@@ -0,0 +1,196 @@
+import static javax.ws.rs.core.MediaType.APPLICATION_OCTET_STREAM;
+import static javax.ws.rs.core.Response.noContent;
+
+@Path( "/api/archive" )
+public class ArchiveManageResources
+{
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    @Inject
+    HistoricalContentListReader reader;
+
+    @Inject
+    ArchiveController controller;
+
+    @Inject
+    ResponseHelper responseHelper;
+
+    @Operation( description = "Generate archive based on tracked content" )
+    @APIResponse( responseCode = "201", description = "The archive is created successfully" )
+    @RequestBody( description = "The tracked content definition JSON", name = "body", required = true,
+            content = @Content( schema = @Schema( implementation = HistoricalContentDTO.class ) ) )
+    @POST
+    @Path( "generate" )
+    @Consumes( APPLICATION_JSON )
+    public Response create( final @Context UriInfo uriInfo, final @Context HttpRequest request )
+    {
+        HistoricalContentDTO content;
+        try
+        {
+            String json = IOUtils.toString( request.getInputStream(), Charset.defaultCharset() );
+            content = objectMapper.readValue( json, HistoricalContentDTO.class );
+            if ( content == null )
+            {
+                final String message = "Failed to read historical content which is empty.";
+                logger.error( message );
+                return responseHelper.fromResponse( message );
+            }
+        }
+        catch ( final IOException e )
+        {
+            final String message = "Failed to read historical content file from request body.";
+            logger.error( message, e );
+            return responseHelper.fromResponse( message );
+        }
+
+        try
+        {
+            Map<String, String> downloadPaths = reader.readPaths( content );
+            controller.downloadArtifacts( downloadPaths, content );
+            Optional<File> archive = controller.generateArchive( content );
+            if ( archive.isEmpty() )
+            {
+                final String message = "Failed to get downloaded contents for archive.";
+                logger.error( message );
+                return responseHelper.fromResponse( message );
+            }
+            controller.renderArchive( archive.get(), content.getBuildConfigId() );
+        }
+        catch ( InterruptedException e )
+        {
+            final String message = "Artifacts downloading is interrupted.";
+            logger.error( message, e );
+            return responseHelper.fromResponse( message );
+        }
+        catch ( final ExecutionException e )
+        {
+            final String message = "Artifacts download execution manager failed.";
+            logger.error( message, e );
+            return responseHelper.fromResponse( message );
+        }
+        catch ( final IOException e )
```

Comment:
can we do a catch like catch(ExecutionException | IOException e)?

Reply:
if we need do different message recording we can do something like 
catch(ex:Exception) {
    when(ex) {
        is SomeException,
        is AnotherException -> {
            // handle
        }
        else -> throw ex
    }

Reply:
@shokakucarrier 
The two exceptions are cathing different phases, download->ExcetionException, generate->IOException
The exception code way you wrote, I need to clarify it's not used generally in Java.